cellarium/ml/train/training_plan.py

```diff
@@ -67,6 +78,98 @@ def __init__(
         self.scheduler_fn = scheduler_fn
         self.scheduler_kwargs = scheduler_kwargs
 
+        if config is not None:
+            self._set_hparams(config)
+
+    @classmethod
+    @patch("lightning.pytorch.core.saving._load_state", new=_load_state)
```

Comment:
I just learned something new! This mocking library is pretty awesome! It's funny that it lives in unittest. Is that a rare hack to use mocking for patching actual code, or a commonly used workaround?

Reply:
Yeah it is pretty neat, found it on stackexchange.

> Is that a rare hack to use mocking for patching actual code, or a commonly used workaround?
Honestly, I don't know. But it seems to do exactly what I want.